### DISCUSSION TITLE
What is "port-index" intended to describe?

### DISCUSSION THREAD

**Discussion by galenlynch** [AllenNeuralDynamics/aind-data-schema#845] [Q&A]
I have been wondering what [`port_index`](https://github.com/AllenNeuralDynamics/aind-data-schema/blob/23ac5fd72eddadffa848e2ae00b154feead93a37/src/aind_data_schema/models/devices.py#L256) of a `Device` is supposed to describe. Does it describe which port of some other device that the described device plugs into?

If so, how should we note the port of a Neuropixels probe, which is commonly described by the basestation port that the twisted pair plugs into, but in reality plugs into a dock on the headstage? Should the `port_index` of the Neuropixels probe be the headstage dock, and the `port_index` of the headstage be the port of the basestation?

**Comment by bruno-f-cruz** [comment]
It came about when I asked to have a way to refer to a concept of hardware index. For cameras it could be index, for serial devices a com port, for network devices an IP, etc... 
Given the already existing schema, the decision of leaving it as a string was to exactly allow users to write down what they decided was the most appropriate, and try to be the least committed to a specific format as possible. If needed, additional validators can be added to the string once the Device class is inherited. 
If the metadata you want to keep track of is specific to your specific device and it follows a very strict structure you should sub classing Device and just add more fields.
